apm-agent-plugins/apm-httpclient11-plugin/src/main/java/co/elastic/apm/agent/httpclient/HttpClientAsyncInstrumentation.java

```diff
@@ -0,0 +1,80 @@
+package co.elastic.apm.agent.httpclient;
+
+import co.elastic.apm.agent.httpclient.helper.AsyncCallbackCreator;
+import co.elastic.apm.agent.httpclient.helper.AsyncCallbackCreatorImpl;
+import co.elastic.apm.agent.httpclient.helper.HttpClientHelper;
+import co.elastic.apm.agent.impl.transaction.AbstractSpan;
+import co.elastic.apm.agent.impl.transaction.Span;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+import javax.annotation.Nullable;
+import java.net.URI;
+import java.net.http.HttpRequest;
+import java.util.concurrent.CompletableFuture;
+
+import static net.bytebuddy.matcher.ElementMatchers.hasSuperType;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.returns;
+import static net.bytebuddy.matcher.ElementMatchers.takesArguments;
+
+public class HttpClientAsyncInstrumentation extends AbstractHttpClientInstrumentation {
+
+    @Override
+    public Class<?> getAdviceClass() {
+        return HttpClient11Advice.class;
+    }
+
+    public static class HttpClient11Advice {
+
+        @Advice.OnMethodEnter(suppress = Throwable.class, inline = false)
+        public static void onBeforeExecute(@Advice.Argument(value = 0) HttpRequest httpRequest) {
+            if (tracer.getActive() == null) {
+                return;
+            }
+            final AbstractSpan<?> parent = tracer.getActive();
+            URI uri = httpRequest.uri();
+            Span span = co.elastic.apm.agent.http.client.HttpClientHelper.startHttpClientSpan(parent, httpRequest.method(), uri.toString(), uri.getScheme(),
+                HttpClientHelper.computeHostName(uri.getHost()), uri.getPort());
+            if (span != null) {
+                span.activate();
+            }
+        }
+
+        @Advice.OnMethodExit(suppress = Throwable.class, onThrowable = Throwable.class, inline = false)
+        public static void onAfterExecute(@Advice.Return @Nullable CompletableFuture completableFuture,
+                                          @Advice.Thrown @Nullable Throwable t) {
+            final Span activeSpan = tracer.getActiveExitSpan();
+            if (activeSpan == null) {
+                return;
+            }
+            activeSpan.deactivate();
+            AsyncCallbackCreator callbackCreator = new AsyncCallbackCreatorImpl();
```

Comment:
These creator interfaces are not needed anymore with indy plugins :)
Just directly implement the BiConsumer as a lambda.

Reply:
fixed